include/engine/plugins/trip.hpp

```diff
@@ -291,13 +287,14 @@ template <class DataFacadeT> class RoundTripPlugin final : public BasePlugin
         {
             const auto component_size = scc.range[k + 1] - scc.range[k];
 
-            BOOST_ASSERT_MSG(component_size >= 0, "invalid component size");
+            BOOST_ASSERT_MSG(component_size > 0, "invalid component size");
+
+            std::vector<NodeID> scc_route;
```

Comment:
`route_results` is a `vector<vector<>>`, so `.emplace_back` would only construct a `vector<>` inplace, and as far as I can see, there is no way a `vector<>` could be meaningfully initialized via `BruteForceTrip()`.

`vector<vector<>>` are not perfect anyway, since they potentially over allocate in both dimensions (fortunately we can `.reserve` here so that's not too bad)